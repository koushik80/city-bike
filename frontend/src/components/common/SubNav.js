import React from 'react';
import { NavLink } from 'react-router-dom';
import './SubNav.scss';

const SubNav = () => {
  return (
    <header className="header-area">
        <div className="header-top">
            <div className="container-fluid">
                <div className="top_nav_area">
                    <div className="col-lg-6">
                        <div className="header_top_left_side">
                            <ul>
                                <li><a href="https://www.hel.fi/en/" target="_blank" rel="noopener noreferrer">Helsinki</a></li>
                                <li><a href="https://www.hsl.fi/" target="_blank" rel="noopener noreferrer">HSL</a></li>
                                <li><NavLink to="/map">Map</NavLink></li>
                            </ul>
                        </div>
                    </div>
                    <div className="col-lg-6">
                        <div className="header-right-social text-center text-sm-right">
                            <ul>
                              <li><a href="https://www.facebook.com/" target="_blank" rel="noopener noreferrer"><i className="fab fa-facebook-f"></i></a></li>
                              <li><a href="https://twitter.com/" target="_blank" rel="noopener noreferrer"><i className="fab fa-twitter"></i></a></li>
                              <li><a href="https://github.com/koushik80/" target="_blank" rel="noopener noreferrer"><i className="fab fa-github"></i></a></li>
                              <li><a href="https://www.linkedin.com/" target="_blank" rel="noopener noreferrer"><i className="fab fa-linkedin-in"></i></a></li>
                              <li><a href="https://www.youtube.com/" target="_blank" rel="noopener noreferrer"><i className="fab fa-youtube"></i></a></li>
                            </ul>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    </header>
  )
}

export default SubNav;