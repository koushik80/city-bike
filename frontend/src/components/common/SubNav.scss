@import "./../../scss/partials/mixins";

.header-area {
  overflow: hidden;

  .header-top {
    position: relative;
    background: #5670ff;
    padding-right: 30px;
    padding-left: 30px;
  }
}

.top_nav_area {
  display: flex;
  flex-wrap: wrap;
  > div {
    width: 50%;
  }

.header_top_left_side {
  position: relative;

  &::before {
    position: absolute;
    content: "";
    height: 100%;
    width: 50%;
    background: white;
    top: 0;
    right: -70px;
    @include transform(skew(-45deg));
    }

    ul {
      display: block;
      li {
        display: inline-block;

        a {
          color: white;
          line-height: 40px;
          margin-right: 40px;
          font-size: 14px;
        }
      }
    }
  }
}
.header-right-social {
  position: relative;
  text-align: right !important;
  &::before {
    position: absolute;
    content: "";
    height: 100%;
    width: 50%;
    background: white;
    top: 0;
    left: -70px;
    @include transform(skew(45deg));
  }

  ul {
    display: block;
    li {
      display: inline-block;

      a {
        color: white;
        margin-left: 24px;
        line-height: 40px;
        display: inline-block;
      }
    }
  }
}

@media screen and (max-width: 792px) {
  .top_nav_area {
    display: flex;
    flex-wrap: wrap;

    .header_top_left_side {
      position: relative;

      &::before {
        width: 0%;
      }
    }
  }

  .header-right-social {
    &::before {
      width: 0%;
    }
  }
}

@media screen and (max-width: 662px) {
  .top_nav_area {
    display: block;
    flex-wrap: wrap;

    > div {
      width: 100%;
    }
    .header_top_left_side {
      position: relative;
    }
  }
}

@media screen and (max-width: 345px) {
  .top_nav_area {
    display: none;
    flex-wrap: wrap;

    > div {
      width: 100%;
    }
    .header_top_left_side {
      position: relative;
    }
  }
}