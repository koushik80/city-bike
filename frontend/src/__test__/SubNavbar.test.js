import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SubNav from '../components/common/SubNav';


describe("Home SubNav", () => {

    it("should render correctly", () => {
        const wrapper = renderer.create(<BrowserRouter><SubNav /></BrowserRouter> );

        expect(wrapper).toMatchSnapshot();
    });
});