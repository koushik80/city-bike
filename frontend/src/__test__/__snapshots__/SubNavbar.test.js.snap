// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home SubNav should render correctly 1`] = `
<header
  className="header-area"
>
  <div
    className="header-top"
  >
    <div
      className="container-fluid"
    >
      <div
        className="top_nav_area"
      >
        <div
          className="col-lg-6"
        >
          <div
            className="header_top_left_side"
          >
            <ul>
              <li>
                <a
                  href="https://www.hsl.fi/en/citybikes/helsinki/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  HSL
                </a>
              </li>
              <li>
                <a
                  className=""
                  href="/map"
                  onClick={[Function]}
                >
                  Map
                </a>
              </li>
              <li>
                <a
                  className=""
                  href="/navigation"
                  onClick={[Function]}
                >
                  cNav
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="col-lg-6"
        >
          <div
            className="header-right-social text-center text-sm-right"
          >
            <ul>
              <li>
                <a
                  href="https://www.facebook.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fab fa-facebook-f"
                  />
                </a>
              </li>
              <li>
                <a
                  href="https://twitter.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fab fa-twitter"
                  />
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/koushik80/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fab fa-github"
                  />
                </a>
              </li>
              <li>
                <a
                  href="https://www.linkedin.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fab fa-linkedin-in"
                  />
                </a>
              </li>
              <li>
                <a
                  href="https://www.youtube.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  <i
                    className="fab fa-youtube"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
